pkg/proxy/iptables/proxier_test.go

```diff
@@ -1470,20 +1787,12 @@ func Test_buildNewEndpointsMap(t *testing.T) {
 		expectedHealthchecks: map[types.NamespacedName]int{},
 	}, {
 		// Case[2]: no change, named port, local
-		newEndpoints: makeEndpointsMap(
-			makeTestEndpoints("ns1", "ep1", func(ept *api.Endpoints) {
-				ept.Subsets = []api.EndpointSubset{{
-					Addresses: []api.EndpointAddress{{
-						IP:       "1.1.1.1",
-						NodeName: &nodeName,
-					}},
-					Ports: []api.EndpointPort{{
-						Name: "p11",
-						Port: 11,
-					}},
-				}}
-			}),
-		),
+		previousEndpoints: []*api.Endpoints{
+			makeTestEndpoints("ns1", "ep1", namedPortLocal),
```

Comment:
I think it is unfortunate to out-of-line these functions.  I don't see an obvious better answer, though